### PR TITLE
Search endpoint frontend

### PR DESCRIPTION
**Pull request by michielderoos**
**Describe the Pull Request**

Frontend for the serverside credit transfer search!   

**Todo**

- [x] Link relevant [trello card](https://trello.com/b/PA2LhlOh/sempo-dev) or [related issue](https://github.com/teamsempo/SempoBlockchain/issues) to the pull request
- [x] Request review from relevant @person or team
- [x] QA your pull request. This includes running the app, login, testing changes, checking [accessibility](https://www.a11yproject.com/checklist/) etc.
- [x] Bump [backend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/config.py#L7) and/or [frontend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/app/package.json#L3) versions following Semver
- [ ] Add release notes to [unreleased changelog](https://github.com/teamsempo/SempoBlockchain/blob/master/CHANGELOG.md#unreleased)

### CODE REVIEW

**Comment by michielderoos** [comment]
[Credit transfer search frontend](https://trello.com/c/j8xRs8AH/859-credit-transfer-search-frontend)

**Review by tristanhcole — COMMENTED** [review]
Overall, looks good! A few small issues, mostly typing and naming, on the frontend. 

This PR breaks the single transfer account page.
Screen Shot 2021-07-06 at 1 24 02 pm

File: app/client/components/pages/dashboardPage.jsx [hunk 1]
```diff
@@ -48,13 +48,9 @@ class DashboardPage extends React.Component {
...
   componentDidMount() {
     this.subscribe();
-
-    let transfer_type = "ALL";
-    let per_page = 50;
+    let per_page = 10;
     let page = 1;
     this.props.loadCreditTransferList({
-      get_stats: true,
-      transfer_type: transfer_type,
```
[the remark this answers is not in the document]
Comment (michielderoos) [reply app/client/components/pages/dashboardPage.jsx:57]: Yep! Replaced by the filters module

File: app/client/components/user/EditUserForm.tsx [hunk 2]
```diff
@@ -386,7 +386,9 @@ class EditUserForm extends React.Component<
             </Wrapper>
           </ModuleBox>
 
-          {Object.keys(customAttributes).length >= 1 || businessUsageName ? (
+          {Object.keys(customAttributes || {}).length >= 1 ||
+          businessUsageName ||
+          "" ? (
```
[the remark this answers is not in the document]
Comment (michielderoos) [reply app/client/components/user/EditUserForm.tsx:391]: Yeah-- customAttributes could be `null`, and that broke `Object.keys`

File: app/client/reducers/creditTransfer/reducers.ts [hunk 3]
```diff
@@ -28,7 +31,7 @@ export const byId = (state = {}, action: CreditTransferAction) => {
           }
         }
       });
-      return DEEEEEEP(state, action.payload);
+      return DEEEEEEP({}, action.payload);
```
[the remark this answers is not in the document]
Comment (michielderoos) [reply app/client/reducers/creditTransfer/reducers.ts:34]: I did this to flush the previous pages' credit transfers. With `return DEEEEEEP(state, action.payload);`, every time you go to the next page in the transfer list page the list just gets longer (I.e. if you load page 1, 2, and 3, then you get one long page with 30 transfers)

File: app/client/sagas/creditTransferSagas.ts [hunk 4]
```diff
@@ -41,8 +43,10 @@ function* updateStateFromCreditTransfer(result: CreditLoadApiResult) {
   } else {
     credit_transfer_list.push(result.data.credit_transfer);
   }
-
-  const normalizedData = normalize(credit_transfer_list, creditTransferSchema);
+  const normalizedData = normalize(
+    credit_transfer_list.reverse(),
```
[the remark this answers is not in the document]
Comment (michielderoos) [reply app/client/sagas/creditTransferSagas.ts:45]: I couldn't quite work out why, but credit transfer list was being returned in the correct order by the BE, but `result.data.credit_transfers` is in reverse order.

File: app/client/components/user/EditUserForm.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (tristanhcole) [reply app/client/components/user/EditUserForm.tsx:391]: makes sense!

File: app/client/reducers/creditTransfer/reducers.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (tristanhcole) [reply app/client/reducers/creditTransfer/reducers.ts:34]: Ok, that makes sense!

File: app/client/sagas/creditTransferSagas.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (tristanhcole) [reply app/client/sagas/creditTransferSagas.ts:45]: strange, but ok

**Review by tristanhcole — APPROVED** [review]
Tested & good to go! 🚀

File: app/client/components/creditTransfer/CreditTransferList.tsx [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+      return <p style={{ margin: 0 }}>{money}</p>;
+    }
+  },
+  {
+    title: "Sender",
+    key: "sender_name",
+    ellipsis: true,
+    render: (text: any, record: any) => (
+      <Link
+        to={"/users/" + record.sender_user_id}
```
Comment (tristanhcole) [inline app/client/components/creditTransfer/CreditTransferList.tsx:100]: nit: potential small change for usability. Have this go to `/accounts/` + transfer_account.id. As it breaks the breadcrumb flow
Screen Shot 2021-07-07 at 8 17 10 am

File: app/client/components/creditTransfer/CreditTransferList.tsx [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+      </Link>
+    )
+  },
+  {
+    title: "Recipient",
+    key: "recipient_name",
+    ellipsis: true,
+    render: (text: any, record: any) => (
+      <Link
+        to={"/users/" + record.recipient_user_id}
```
Comment (tristanhcole) [inline app/client/components/creditTransfer/CreditTransferList.tsx:117]: See https://github.com/teamsempo/SempoBlockchain/pull/726/files#r664924108

**Pull request merged**